docs/core/compatibility/extensions/10.0/getkeyedservice-anykey.md

```diff
@@ -0,0 +1,83 @@
+---
+title: "Breaking change: Fix issues in GetKeyedService() and GetKeyedServices() with AnyKey"
+description: "Learn about the breaking change in .NET 10 Preview 3 where GetKeyedService() and GetKeyedServices() behavior changed when using KeyedService.AnyKey as the lookup key."
+ms.date: 11/19/2025
+ai-usage: ai-assisted
+---
+
+# Fix issues in GetKeyedService() and GetKeyedServices() with AnyKey
+
+The behavior of `GetKeyedService()` and `GetKeyedServices()` methods in the `Microsoft.Extensions.DependencyInjection` library was updated to address inconsistencies in handling the `KeyedService.AnyKey` registration. Specifically, `GetKeyedService()` now throws an exception when you attempt to resolve a single service using `KeyedService.AnyKey` as the lookup key, and `GetKeyedServices()` no longer returns `AnyKey` registrations when queried with `KeyedService.AnyKey`.
+
+## Version introduced
+
+.NET 10 Preview 3
+
+## Previous behavior
+
+Previously, calling `GetKeyedService()` with `KeyedService.AnyKey` would return a service registration associated with `AnyKey`. This behavior was inconsistent with the intended semantics, as `AnyKey` is meant to represent a special case of keyed services rather than a specific registration.
+
+Additionally, calling `GetKeyedServices()` with `KeyedService.AnyKey` would return all registrations for `AnyKey`. This behavior was also inconsistent with the intended semantics, as `AnyKey` is not meant to enumerate all keyed services.
```

Comment:
```suggestion
Previously, calling `GetKeyedService()` with `KeyedService.AnyKey` returned a service registration associated with `AnyKey`. This behavior was inconsistent with the intended semantics, as `AnyKey` is meant to represent a special case of keyed services rather than a specific registration.

Calling `GetKeyedServices()` with `KeyedService.AnyKey` returned all registrations for `AnyKey`. This behavior was also inconsistent with the intended semantics, as `AnyKey` is not meant to enumerate all keyed services.
```